.github/workflows/PR-build.yml

```diff
@@ -0,0 +1,144 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: PR Build
+on:
+  pull_request:
+    branches: 
+      - main 
+    types: 
+      - opened
```

Comment:
do we need to execute this on opened if we already have it on ready_for_review

Reply:
Opened is used for draft PR based on my understanding. However, I'm fine with using `ready_for_review` since it shows the final state and is the final gate before merging the PR.